### PR TITLE
fix(client): sci-py layout to grid

### LINKED ISSUE #52664: Python Projects need to be a grid
They look like this:

image

They need to look like this:

image

But doing so makes the cert projects look like this:

image

---

We may need to give each of the cert projects their own blocks. Open to other solutions, as well.

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [ ] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [ ] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [ ] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or GitPod.

Closes #52664

### CODE REVIEW

File: client/src/utils/is-a-cert.ts [hunk 1]
```diff
@@ -19,3 +19,7 @@ export function isRelationalDbCert(superBlock: string): boolean {
 export function isCollegeAlgebraPyCert(superBlock: string): boolean {
   return superBlock === String(SuperBlocks.CollegeAlgebraPy);
 }
+
+export function isNewPythonCert(superBlock: string): boolean {
+  return superBlock === String(SuperBlocks.SciCompPy);
+}
```
Comment (ShaunSHamilton) [inline client/src/utils/is-a-cert.ts:25]:
```suggestion
export function isSciCompPyCert(superBlock: string): boolean {
  return superBlock === String(SuperBlocks.SciCompPy);
}
```

File: client/src/templates/Introduction/components/challenges.tsx [hunk 2]
```diff
@@ -33,7 +37,11 @@ function Challenges({
 }: Challenges): JSX.Element {
   const { t } = useTranslation();
 
-  const isGridMap = isNewRespCert(superBlock) || isNewJsCert(superBlock);
+  const isGridMap = [
+    isNewRespCert(superBlock),
+    isNewJsCert(superBlock),
+    isNewPythonCert(superBlock)
+  ].some(truthy => truthy);
```
Comment (ShaunSHamilton) [inline client/src/templates/Introduction/components/challenges.tsx:44]:
```suggestion
    isSciCompPyCert(superBlock) && !isProjectBlock
  ].some(Boolean);
```

File: client/src/templates/Introduction/components/challenges.tsx [hunk 3]
```diff
@@ -10,7 +10,11 @@ import GreenPass from '../../../assets/icons/green-pass';
 import { executeGA } from '../../../redux/actions';
 import { SuperBlocks } from '../../../../../shared/config/superblocks';
 import { ChallengeWithCompletedNode } from '../../../redux/prop-types';
-import { isNewJsCert, isNewRespCert } from '../../../utils/is-a-cert';
+import {
+  isNewJsCert,
+  isNewPythonCert,
```
Comment (ShaunSHamilton) [inline client/src/templates/Introduction/components/challenges.tsx:15]:
```suggestion
  isSciCompPyCert,
```

File: client/src/templates/Introduction/components/block.tsx [hunk 4]
```diff
@@ -107,6 +108,7 @@ class Block extends Component<BlockProps> {
     const isNewResponsiveWebDesign = isNewRespCert(superBlock);
     const isNewJsAlgos = isNewJsCert(superBlock);
     const isOdinProject = blockDashedName == 'the-odin-project';
+    const isNewPython = isNewPythonCert(superBlock);
```
Comment (ShaunSHamilton) [inline client/src/templates/Introduction/components/block.tsx:111]:
```suggestion
    const isSciCompPy = isSciCompPyCert(superBlock);
```

File: client/src/templates/Introduction/components/block.tsx [hunk 5]
```diff
@@ -22,6 +22,7 @@ import { makeExpandedBlockSelector, toggleBlock } from '../redux';
 import {
   isCollegeAlgebraPyCert,
   isNewJsCert,
+  isNewPythonCert,
```
Comment (ShaunSHamilton) [inline client/src/templates/Introduction/components/block.tsx:25]:
```suggestion
  isSciCompPyCert,
```

File: client/src/templates/Introduction/components/block.tsx [hunk 6]
```diff
@@ -352,23 +354,23 @@ class Block extends Component<BlockProps> {
       </ScrollableAnchor>
     );
 
+    const shouldBeGrid = [
+      isNewResponsiveWebDesign,
+      isNewJsAlgos,
+      isCollegeAlgebraPy,
+      isNewPython
+    ].some(truthy => truthy);
```
Comment (ShaunSHamilton) [inline client/src/templates/Introduction/components/block.tsx:362]:
```suggestion
  const shouldBeGrid = [
    isNewRespCert(superBlock),
    isNewJsCert(superBlock),
    isCollegeAlgebraPy,
    isSciCompPyCert(superBlock) && !isProjectBlock
  ].some(Boolean);
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
I much prefer the name being related to what is being tested.

**Comment by ShaunSHamilton** [comment]
PS: I had started working on this, but went with this approach:

```ts
    const blockrenderer = () => {
      if (isProjectBlock && !isOdinProject) {
        switch (true) {
          case isNewResponsiveWebDesign:
          case isNewJsAlgos:
          case isCollegeAlgebraPy:
            return GridProjectBlock;
          default:
            return ProjectBlock;
        }
      } else {
        switch (true) {
          case isNewResponsiveWebDesign:
          case isNewJsAlgos:
          case isCollegeAlgebraPy:
          case isSciCompPy:
            return GridBlock;
          default:
            return Block;
        }
      }
    };
```

**Comment by ShaunSHamilton** [comment]
I do not know if we want this like it is now:
image

Or like this:
image

**Comment by Sembauke** [comment]
I am fine with both of them.

**Review by ShaunSHamilton — APPROVED** [review]
Well, I did not see anything broken. Nice one, Sem!

**Review by huyenltnguyen — APPROVED** [review]

**Pull request merged**